Src/WindowsServer/WindowsServer.Shared/AppServicesHeartbeatTelemetryModule.cs

```diff
@@ -0,0 +1,181 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.WindowsServer.Implementation;
+
+    /// <summary>
+    /// Provides default values for the heartbeat feature of Application Insights that
+    /// are specific to Azure App Services (Web Apps, Functions, etc...).
+    /// </summary>
+    public sealed class AppServicesHeartbeatTelemetryModule : ITelemetryModule, IDisposable
+    {
+        /// <summary>
+        /// Environment variables and the Application Insights heartbeat field names that accompany them.
+        /// </summary>
+        internal static readonly KeyValuePair<string, string>[] WebHeartbeatPropertyNameEnvVarMap = new KeyValuePair<string, string>[]
+        {
+            new KeyValuePair<string, string>("appSrv_SiteName", "WEBSITE_SITE_NAME"),
```

Comment:
Aren't these values appended to all telemetry items by the corresponding TelemetryInitializers? Why do we send these values (duplicated) into HeartBeat again?

Reply:
Heartbeat and telemetry initializers are two separate solutions. I am not 100% aware of all the downstream consumers of the TelemetryInitializers, but I am keenly aware of the heartbeat ones (app 2 infra). I want the app 2 infra feature to only look in 1 single place for the values it requires. I believe the duplication caused by heartbeats (default 15-minute cadence) is minimal enough to support the end purpose.